cmd/vic-machine/list/list.go

```diff
@@ -0,0 +1,141 @@
+// Copyright 2016 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package list
+
+import (
+	"path"
+	"text/tabwriter"
+	"text/template"
+
+	log "github.com/Sirupsen/logrus"
+
+	"github.com/urfave/cli"
+	"github.com/vmware/vic/lib/install/data"
+	"github.com/vmware/vic/lib/install/management"
+	"github.com/vmware/vic/lib/install/validate"
+	"github.com/vmware/vic/pkg/errors"
+	"github.com/vmware/vic/pkg/trace"
+	"github.com/vmware/vic/pkg/vsphere/vm"
+
+	"golang.org/x/net/context"
+)
+
+type items struct {
+	ID   string
+	Path string
+	Name string
+}
+
+const templ = `{{range .}}
+{{.ID}}	{{.Path}}	{{.Name}}{{end}}
+`
+
+// Delete has all input parameters for vic-machine delete command
+type List struct {
+	*data.Data
+
+	executor *management.Dispatcher
+}
+
+func NewList() *List {
+	d := &List{}
+	d.Data = data.NewData()
+	return d
+}
+
+// Flags return all cli flags for delete
+func (l *List) Flags() []cli.Flag {
+	flags := append(l.TargetFlags(), l.ComputeFlagsNoName()...)
+	flags = append(flags, l.DebugFlags()...)
+
+	return flags
+}
+
+func (l *List) processParams() error {
+	defer trace.End(trace.Begin(""))
+
+	if err := l.HasCredentials(); err != nil {
+		return err
+	}
+
+	l.Insecure = true
+	return nil
+}
+
+func (l *List) prettyPrint(cli *cli.Context, ctx context.Context, vchs []*vm.VirtualMachine) {
+	data := []items{
```

Comment:
I know that in your design, there is one Notes column, but as we still didn't support that argument in create, I didn't add that column in ls command as well.
Should we just add one column here, but leave it empty? Or you'd like to give one suggested argument, to add into create command?